lib/rules/no-shared-component-data.js

```diff
@@ -21,10 +27,33 @@ function create (context) {
         p.value.type !== 'ArrowFunctionExpression' &&
         p.value.type !== 'Identifier'
       )
-      .forEach(cp => {
+      .forEach(p => {
         context.report({
-          node: cp.value,
-          message: '`data` property in component must be a function'
+          node: p,
+          message: '`data` property in component must be a function',
+          fix (fixer) {
+            let first = sourceCode.getFirstToken(p.value)
+            let last = sourceCode.getLastToken(p.value)
+
+            // If the value enclosed by parentheses, update the 'first' and 'last' by the parentheses.
+            let t = null
+            let u = null
+            while (isOpenParen(t = sourceCode.getTokenBefore(first)) && isCloseParen(u = sourceCode.getTokenAfter(last))) {
+              first = t
+              last = u
+            }
+
+            // If we can upgrade requirements to `eslint@>4.1.0`, this code can be replaced by:
+            //     return [
+            //       fixer.insertTextBefore(first, 'function() {\nreturn '),
+            //       fixer.insertTextAfter(last, ';\n}')
+            //     ]
+            // See: https://eslint.org/blog/2017/06/eslint-v4.1.0-released#applying-multiple-autofixes-simultaneously
+            const range = [first.range[0], last.range[1]]
+            const valueText = sourceCode.text.slice(range[0], range[1])
+            const replacement = `function() {\nreturn ${valueText};\n}`
```

Comment:
Can't we replace it with `data() {}` instead of a `function`?

Reply:
I think that a function expression is better to keep the rule simple.
If we want to fix to a method shorthand, it needs extra check the `object-shorthand` rule is doing. I.e., if comments exist between the key and the value, the rule should not autofix it to avoid removing the comments.

Reply:
Ah, alright. There is already `object-shorthand` core rule, that is fixable, so it should take care of it :) I forgot about it.